contract/src/launchIt.js

```diff
@@ -0,0 +1,255 @@
+// @ts-check
+import { E, Far } from '@endo/far';
+import { M, mustMatch } from '@endo/patterns';
+import {
+  BrandShape,
+  AmountShape,
+  AssetKindShape,
+  DisplayInfoShape,
+} from '@agoric/ertp/src/typeGuards.js';
+import { AmountMath, AssetKind } from '@agoric/ertp/src/amountMath.js';
+import {
+  TimerServiceShape,
+  TimestampRecordShape,
+  TimestampValueShape,
+} from '@agoric/time/src/typeGuards.js';
+import { atomicRearrange } from '@agoric/zoe/src/contractSupport/index.js';
+import { makeRatio, multiplyBy } from '@agoric/zoe/src/contractSupport/ratio';
+
+const { Fail, quote: q } = assert;
+
+/** @type {import('./types').ContractMeta} */
+export const meta = {
+  customTermsShape: M.splitRecord(
+    { name: M.string(), supplyQty: M.bigint() },
+    { displayInfo: DisplayInfoShape },
+  ),
+};
+export const { customTermsShape } = meta;
+/**
+ * @typedef {{
+ *   name: string,
+ *   supplyQty: bigint,
+ *   displayInfo?: DisplayInfo,
+ * }} LaunchTerms
+ */
+
+const kw = s => (/^[A-Z]/.test(s) ? s : `KW${s}`); // only addresses initial cap
+
+/**
+ * @param {ZCF} zcf
+ * @param {string} kw
+ */
+const makeZcfIssuerKit = async (zcf, kw) => {
+  const mint = await zcf.makeZCFMint(kw);
+  const kit = mint.getIssuerRecord();
+  return { ...kit, mint };
+};
+
+/**
+ * This contract is limited to fungible assets.
+ *
+ * @param {ZCF<LaunchTerms>} zcf
+ * @param {unknown} _privateArgs
+ * @param {import('@agoric/vat-data').Baggage} _baggage
+ */
+export const start = async (zcf, _privateArgs, _baggage) => {
+  const { name, supplyQty, displayInfo = {}, brands } = zcf.getTerms();
+  mustMatch(brands, M.splitRecord({ Deposit: BrandShape }));
```

Comment:
Why `M.splitRecord(...)`  instead of `{  Deposit: BrandShape }`? Are we expecting `brands` to be keyed with anything else?

Reply:
At the time, I only needed this brand to exist for the code I was writing to be correct.

Stepping back and looking at the whole thing now, no, I don't think there are any others.